app/app/src/main/kotlin/com/hedvig/android/app/MainActivity.kt

```diff
@@ -127,6 +121,7 @@ class MainActivity : AppCompatActivity() {
       }
     }
 
+    val externalNavigator = ExternalNavigatorImpl(this, hedvigBuildConstants.appId)
```

Comment:
Definitely not a dumb question.
We want to have the Activity instance in there, which is something I am not 100% sure how safe it is to do through Koin, or if it's possible at all. I know it knows how to inject the Application context, but not feel confident with the activity itself. So I just took the simplest approach which is to just do this here instead.

Also we do not want to pass ExternalNavigator to anything like a Repository or a UseCase I don't think at least. It's mostly things that we do from our composables instead. So it is a good thing to not have it in koin so if some of them tries to inject it it will simply crash.
If what I am saying here turns out to be false and we do want to do that sometime in the future we should evaluate it when that time comes.

Reply:
I think it would require to have some ActivityProvider class with currentActivity changed on application.registerActivityLifecycleCallbacks. This is definitely simpler:)